scipy/stats/_kde.py

```diff
@@ -390,19 +383,13 @@ def integrate_box(self, low_bounds, high_bounds, maxpts=None):
 
         """
         if maxpts is not None:
-            extra_kwds = {'maxpts': maxpts}
-        else:
-            extra_kwds = {}
-
-        with MVN_LOCK:
-            value, inform = _mvn.mvnun_weighted(low_bounds, high_bounds,
-                                                self.dataset, self.weights,
-                                                self.covariance, **extra_kwds)
-        if inform:
-            msg = f'An integral in _mvn.mvnun requires more points than {self.d * 1000}'
-            warnings.warn(msg, stacklevel=2)
+            kwds.update({'maxpts': maxpts})
 
-        return value
+        low, high = low_bounds - self.dataset.T, high_bounds - self.dataset.T
+        values = multivariate_normal.cdf(
+            high, lower_limit=low, cov=self.covariance, **kwds
+        )
+        return (values * self.weights).sum(axis=-1)
```

Comment:
Probably not the bottleneck, but would be nice to use `np.vecdot` when it's available.

Reply:
You're suggesting an explicit branching on the numpy version, or just a comment saying can use vecdot when minimum numpy is >= 2.0?

Reply:
I figure that if it's just a comment, it probably won't happen. I'll bet there are a lot of places that would benefit from vecdot, so maybe a helper in `_util` that does the branching. (It's available pre 2.0 in `numpy.array_api`.)

If you'd prefer, leave it like this, and I'll consider a separate PR to change it here and other places that could use it. It really made a big speed difference in `pearsonr`.

Reply:
FTR, 

```
In [8]: a = np.arange(10000, dtype=float)

In [9]: %timeit (a * a).sum(axis=-1)
7.08 μs ± 35 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

In [10]: %timeit np.vecdot(a, a)
2.39 μs ± 18.9 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```

Wow. In retrospect, it's not that surprising: probably `O(N)` copying to a temp array for the product takes longer than `O(N)` work. Still, did not expect this.

Let me add a comment to this effect so that we change it everywhere in the array api conversion.

Reply:
I imagined also that `vecdot` can use a single BLAS routine for the operation?

It's OK, I'll just open a PR now to change it in a few places.